### PR TITLE
Service registration

### LINKED ISSUE #152: Add support for service registration configuration
Starting in Vault 1.4, it is now possible to do separate specification for service registration. This is useful for example when using separate Consul clusters for k/v / service discovery.

https://www.vaultproject.io/docs/configuration/service-registration

### PR DESCRIPTION
**Pull request by Legogris**
This adds support for [service registration](https://www.vaultproject.io/docs/configuration/service-registration/consul) configuration.

This should resolve #152 .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -445,11 +560,6 @@ for the DynamoDB storage backend.
   - Override with `VAULT_TLS_SRC_FILES` environment variable
 - Default value: `{{ role_path }}/files`
 
-### `vault_tls_config_path`
```
Comment (bbaassssiiee) [inline README.md:448]: Why was this removed?

Comment (Legogris) [reply README.md:448]: Because there is a duplicate on `:426`.

**Review by bbaassssiiee — APPROVED** [review]
LFTM
